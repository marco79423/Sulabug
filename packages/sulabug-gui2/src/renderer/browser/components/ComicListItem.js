import * as React from 'react'
import {createStyles, withStyles} from '@material-ui/core/styles'
import dateFormat from 'date-fns/format'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import CardActions from '@material-ui/core/CardActions'
import Button from '@material-ui/core/Button'
import CollectionsIcon from '@material-ui/icons/Collections'
import OpenInBrowserIcon from '@material-ui/icons/OpenInBrowser'
import ListItem from '@material-ui/core/ListItem/ListItem'
import {useDispatch, useSelector} from 'react-redux'
import {addComicToCollectionsRequest} from '../ducks/actions'
import {getComicMap} from '../ducks/selectors'

const styles = (theme) => createStyles({
  root: {
    padding: theme.spacing(1),
    '&:not(:first-child)': {
      paddingTop: 0
    }
  },
  card: {
    display: 'flex',
  },
  title: {
    display: 'flex',
    alignItems: 'center',
  },
  metadata: {
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',

    marginTop: theme.spacing(1),
  },
  details: {
    display: 'flex',
    flexDirection: 'column',
    width: 300,
  },
  content: {
    flex: '1 0 auto',
  },
  cover: {
    width: 120,
    height: 150,
  },
  summary: {
    marginTop: theme.spacing(1),
    height: 40,
    textOverflow: 'ellipsis',
    overflow: 'hidden'
  },
  actions: {
    display: 'flex',
    justifyContent: 'flex-end',
  },
  openInBrowserIcon: {
    marginRight: theme.spacing(1),
  },
  collectionsIcon: {
    marginRight: theme.spacing(1),
  },
})

function ComicListItem({classes, comicId}) {
  const dispatch = useDispatch()
  const comicMap = useSelector(getComicMap)
  const comic = comicMap[comicId]

  const openInBrowser = () => {
    require('electron').shell.openExternal(comic.sourcePageUrl)
  }

  const addComicToCollections = () => dispatch(addComicToCollectionsRequest(comicId))

  return (
    <ListItem className={classes.root}>
      <Card className={classes.card}>
        <CardMedia
          className={classes.cover}
          image={comic.coverUrl}
          title={comic.name}
        />
        <div className={classes.details}>
          <CardContent className={classes.content}>
            <Typography className={classes.title} noWrap={true} variant="h2">{comic.name}</Typography>
            <Typography className={classes.metadata} variant="subtitle2" color="textSecondary">
              最新: {comic.lastUpdatedChapter} ({dateFormat(new Date(comic.lastUpdatedTime), 'yyyy/MM/dd')})
            </Typography>
            <Typography variant="subtitle2">{comic.summary.substring(0, 35)}...</Typography>
          </CardContent>
          <CardActions className={classes.actions}>
            <Button size="small" onClick={openInBrowser}><OpenInBrowserIcon
              className={classes.openInBrowserIcon}/><Typography variant="button">試看</Typography></Button>
            <Button size="small" variant="contained" color="primary" disabled={comic.inCollection}
                    onClick={addComicToCollections}><CollectionsIcon
              className={classes.collectionsIcon}/>{comic.inCollection ? '已收藏' : '收藏'}</Button>
          </CardActions>
        </div>
      </Card>
    </ListItem>
  )
}

export default withStyles(styles)(ComicListItem)
