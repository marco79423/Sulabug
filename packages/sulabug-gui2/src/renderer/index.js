import React from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter as Router, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'
import {Provider} from 'react-redux'

import configureStore from './configureStore'
import Browser from './browser'

const store = configureStore()

function App() {
  return (
    <Provider store={store}>
      <Router>
        <Switch>
          <Redirect exact from='/' to='/browser'/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Router>
    </Provider>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
