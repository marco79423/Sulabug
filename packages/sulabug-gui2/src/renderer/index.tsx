import React, {lazy, Suspense} from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'

const Manager = lazy(() => import('./manager'))
const Browser = lazy(() => import('./browser'))

function App() {
  return (
    <HashRouter>
      <Suspense fallback={<div>loading</div>}>
        <Switch>
          <Redirect exact from='/' to='/manager'/>
          <Route path="/manager" component={Manager}/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Suspense>
    </HashRouter>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
