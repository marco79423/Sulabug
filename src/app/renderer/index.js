import * as React from 'react'
import ReactDOM from 'react-dom'

import 'reflect-metadata'

import App from '../../imports/app/App'

ReactDOM.render(
  <App/>,
  document.getElementById('app')
)
