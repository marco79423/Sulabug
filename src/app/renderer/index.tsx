import * as React from 'react'
import {render} from 'react-dom'
import App from '../../imports/ui/pages/App'

render(
  <App/>,
  document.getElementById('app')
)