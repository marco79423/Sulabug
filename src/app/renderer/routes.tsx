import React from 'react'
import { Router, Route, Switch } from 'react-router'
import createBrowserHistory from 'history/createBrowserHistory'

import BrowsePage from '../../imports/ui/pages/App'

const browserHistory = createBrowserHistory()

export const renderRoutes = () => (
  <Router history={browserHistory}>
    <Switch>
      <Route exact path="/" component={BrowsePage}/>
    </Switch>
  </Router>
)
