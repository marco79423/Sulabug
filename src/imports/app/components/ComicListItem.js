import * as React from 'react'
import {createStyles, withStyles} from '@material-ui/core/styles'
import dateformat from 'dateformat'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import CardActions from '@material-ui/core/CardActions'
import Button from '@material-ui/core/Button'
import SaveIcon from '@material-ui/icons/Save'
import ListItem from '@material-ui/core/ListItem/ListItem'


const styles = (theme) => createStyles({
  root: {
    padding: theme.spacing.unit,
    '&:not(:first-child)': {
      paddingTop: 0
    }
  },
  card: {
    display: 'flex',
  },
  title: {
    display: 'flex',
    alignItems: 'center',
  },
  metadata: {
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',

    marginTop: theme.spacing.unit,
  },
  details: {
    display: 'flex',
    flexDirection: 'column',
    width: 300,
  },
  content: {
    flex: '1 0 auto',
  },
  cover: {
    width: 120,
    height: 150,
  },
  summary: {
    marginTop: theme.spacing.unit,
    height: 40,
    textOverflow: 'ellipsis',
    overflow: 'hidden'
  },
  actions: {
    display: 'flex',
    justifyContent: 'flex-end',
  },
  saveIcon: {
    marginRight: theme.spacing.unit,
  },
})

export class ComicListItem extends React.Component {

  createDownloadTask = () => {
    const {comic} = this.props
    this.props.createDownloadTask(comic.id)
  }

  render() {
    const {classes, comic} = this.props
    return (
      <ListItem className={classes.root}>
        <Card className={classes.card}>
          <CardMedia
            className={classes.cover}
            image={comic.coverDataUrl}
            title={comic.name}
          />
          <div className={classes.details}>
            <CardContent className={classes.content}>
              <Typography className={classes.title} noWrap={true} variant="h2">{comic.name}</Typography>
              <Typography className={classes.metadata} variant="subtitle2" color="textSecondary">
                最新: {comic.lastUpdatedChapter} ({dateformat(comic.lastUpdatedTime, 'yyyy/mm/dd')})
              </Typography>
            </CardContent>
            <CardActions className={classes.actions}>
              <Button size="small" onClick={this.createDownloadTask}><SaveIcon className={classes.saveIcon}/><Typography variant="button">下載</Typography></Button>
            </CardActions>
          </div>
        </Card>
      </ListItem>
    )
  }
}

export default withStyles(styles)(ComicListItem)
