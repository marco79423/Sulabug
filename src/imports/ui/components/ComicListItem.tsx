import * as React from 'react'
import {withStyles} from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import CardActions from '@material-ui/core/CardActions'
import Button from '@material-ui/core/Button'
import SaveIcon from '@material-ui/icons/Save'
import ListItem from "@material-ui/core/ListItem/ListItem";


const styles = theme => ({
    root: {
        paddingTop: 0,
        paddingLeft: 0,
        paddingRight: 0,
        paddingBottom: theme.spacing.unit * 2,
    },
    card: {
        display: 'flex',
    },
    subheading: {
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis',
    },
    details: {
        display: 'flex',
        flexDirection: 'column',
    },
    content: {
        flex: '1 0 auto',
    },
    cover: {
        width: 120,
        height: 150,
    },
    actions: {
        display: 'flex',
        justifyContent: 'flex-end',
    },
    saveIcon: {
        marginRight: theme.spacing.unit,
    },
})

interface PropsTypes {
    classes: any,
}

export class ComicListItem extends React.Component<PropsTypes, {}> {
    render() {
        const {classes} = this.props
        return (
            <ListItem className={classes.root}>
                <Card className={classes.card}>
                    <CardMedia
                        className={classes.cover}
                        image="http://rs.sfacg.com/web/novel/images/NovelCover/Small/2017/01/bfb9f16b-bcff-4604-95d3-435d629c20d2.jpg"
                        title="Live from space album cover"
                    />
                    <div className={classes.details}>
                        <CardContent className={classes.content}>
                            <Typography variant="headline">Name</Typography>
                            <Typography className={classes.subheading} variant="subheading" color="textSecondary">
                                Author | SF | Catalog | Status
                            </Typography>
                        </CardContent>
                        <CardActions className={classes.actions}>
                            <Button size="small"><SaveIcon className={classes.saveIcon}/> 下載</Button>
                        </CardActions>
                    </div>
                </Card>
            </ListItem>
        )
    }
}

export default withStyles(styles)(ComicListItem)
