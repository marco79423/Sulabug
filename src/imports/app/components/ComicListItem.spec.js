import 'reflect-metadata'

import React from 'react'
import renderer from 'react-test-renderer'

import ComicListItem from './ComicListItem'


describe('ComicListItem', () => {
  it('will render comic item correctly', async () => {
    const component = renderer.create(
      <ComicListItem
        comic={{
          id: 'id',
          name: 'name',
          coverDataUrl: 'coverDataUrl',
          source: 'source',
          pageUrl: 'pageUrl',
          catalog: 'catalog',
          author: 'author',
          lastUpdated: 'lastUpdated',
          summary: 'summary',
        }}
        createDownloadTask={() => {
        }}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
