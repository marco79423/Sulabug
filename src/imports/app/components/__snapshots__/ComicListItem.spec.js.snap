// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicListItem will render comic item correctly 1`] = `
<li
  className="MuiListItem-root-11 MuiListItem-default-14 MuiListItem-gutters-19 ComicListItem-root-1"
  disabled={false}
>
  <div
    className="MuiPaper-root-24 MuiPaper-elevation1-27 MuiPaper-rounded-25 MuiCard-root-23 ComicListItem-card-2"
  >
    <div
      className="MuiCardMedia-root-51 ComicListItem-cover-7"
      style={
        Object {
          "backgroundImage": "url(\\"coverDataUrl\\")",
        }
      }
      title="name"
    />
    <div
      className="ComicListItem-details-5"
    >
      <div
        className="MuiCardContent-root-53 ComicListItem-content-6"
      >
        <h2
          className="MuiTypography-root-54 MuiTypography-h2-67 MuiTypography-noWrap-80 ComicListItem-title-3"
        >
          name
        </h2>
        <h6
          className="MuiTypography-root-54 MuiTypography-subtitle2-73 MuiTypography-colorTextSecondary-87 ComicListItem-metadata-4"
        >
          最新: 
          lastUpdatedChapter
           (
          2019/01/16
          )
        </h6>
        <h6
          className="MuiTypography-root-54 MuiTypography-subtitle2-73"
        >
          summary
          ...
        </h6>
      </div>
      <div
        className="MuiCardActions-root-90 ComicListItem-actions-9"
      >
        <button
          className="MuiButtonBase-root-119 MuiButton-root-93 MuiButton-text-95 MuiButton-flat-98 MuiButton-sizeSmall-116 MuiCardActions-action-92"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-94"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-122 ComicListItem-collectionsIcon-10"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4l2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
              />
            </svg>
            <span
              className="MuiTypography-root-54 MuiTypography-button-65"
            >
              收藏
            </span>
          </span>
          <span
            className="MuiTouchRipple-root-131"
          />
        </button>
      </div>
    </div>
  </div>
</li>
`;
